install.sh

```diff
@@ -1,7 +1,78 @@
 #!/bin/bash
 
+#repo addresses
+aasdkRepo="https://github.com/OpenDsh/aasdk"
+gstreamerRepo="git://anongit.freedesktop.org/gstreamer/qt-gstreamer"
+openautoRepo="https://github.com/rhysmorgan134/openauto.git"
+
+#Help text
+display_help() {
+    echo
+    echo "   --deps           install all dependencies"
+    echo "   --aasdk          install and build aasdk"
+    echo "   --openauto       install and build openauto "
+    echo "   --gstreamer      install and build gstreamer "
+    echo "   --dash           install and build dash "
+    echo
+    exit 1
+}
+
+#location of OS details for linux
+OS_RELEASE_FILE="/etc/os-release"
+
+#check if Raspian is in the file, if not set the install Args to be false
+if grep -q "Raspbian" ${OS_RELEASE_FILE}; then
+  installArgs="-DRPI_BUILD=true"
+  isRpi=true
+else
+  installArgs=""
+  isRpi=false
+fi
+
+#check to see if there are any arguments supplied, if none are supplied run full install
+if [ $# -gt 0 ]; then
```

Comment:
Definitely better. As a fun exercise, note this alternate way of parsing the command line parameters:

```
args=( "$@" )
valid_args=( deps aasdk gstreamer openauto dash perms )

for option in "${valid_args[@]}"; do
    if [[ "${args[@]}" =~ ${option} ]]; then
       declare $option=true
    fi
done

# blah blah blah

if [ ! -z "$aasdk" ]; then
    # do aasdk stuff
fi
```

This allows you to set the variables externally as environment variables instead of passing them in on the script. If you don't want that you can add `else declare $option=false`.

Reply:
It also opens the door to validating that the given arguments are valid.